t/emma.t

```diff
@@ -0,0 +1,177 @@
+    ok($xc->findnodes(qq(//mets:fptr))->size() == 0);
+  };
+
+  it "uses the emma mets profile" => sub {
+    ok($xc->findnodes('//mets:mets[@PROFILE="http://www.hathitrust.org/documents/hathitrust-emma-mets-profile1.0.xml"]')->size() == 1);
+  };
+
+  it "has a PREMIS message digest calculation event" => sub {
+    ok($xc->findnodes('//premis:eventType[text()="message digest calculation"]')->size() == 1);
+  }
+
+};
+
+context "with volume & temporary ingest/preingest/zipfile dirs" => sub {
+  my $volume;
+  my $objid;
+  my $pt_objid;
+
+  my $test_home;
+  my $tmpdir;
+
+  my $tmpdirs;
+
+  before all => sub {
+    load_db_fixtures;
+    $tmpdirs = HTFeed::Test::TempDirs->new();
+    $objid = "emmatest";
+    $pt_objid = "emmatest";
+  };
+
+  before each => sub {
+    $tmpdirs->setup_example;
+    set_config($tmpdirs->test_home . "/fixtures/emma",'staging','fetch');
+
+    $volume = HTFeed::Volume->new(namespace => 'emma',
+      objid => $objid,
+      packagetype => 'emma');
+  };
+
+  after each => sub {
+    $tmpdirs->cleanup_example;
+  };
+
+  after all => sub {
+    $tmpdirs->cleanup;
+  };
+
+  describe "HTFeed::PackageType::EMMA::Unpack" => sub {
+    my $stage;
+
+    before each => sub {
+      $stage = HTFeed::PackageType::EMMA::Unpack->new(volume => $volume);
+    };
+
+    it "succeeds" => sub {
+      $stage->run();
+      ok($stage->succeeded());
+    };
+
+    it "extracts the zip" => sub {
+      $stage->run();
+      ok(-e "$tmpdirs->{ingest}/$pt_objid/foo.txt");
+    };
+
+    after each => sub {
+      $stage->clean();
+    };
+  };
+
+  describe "HTFeed::PackageType::EMMA::SourceMETS" => sub {
+    my $stage;
+
+    before each => sub {
+      HTFeed::PackageType::EMMA::Unpack->new(volume => $volume)->run();
+
+      $stage = HTFeed::PackageType::EMMA::SourceMETS->new(volume => $volume);
+
+      mock_zephir();
+    };
+
+    it "succeeds" => sub {
+      $stage->run();
+      ok($stage->succeeded());
+    };
+
+    it "generates the METS xml" => sub {
+      $stage->run();
+      ok(-e "$tmpdirs->{ingest}/$pt_objid/$pt_objid.mets.xml");
+    };
+
+    it "caches metadata from the emma xml to the database";
+
+    context "with a mets xml" => sub {
+      share my %mets_vars;
+
+      before each => sub {
+        $stage->run;
+        $mets_vars{xc} = $volume->_parse_xpc("$tmpdirs->{ingest}/$pt_objid/$pt_objid.mets.xml");
+      };
+
+      it_should_behave_like "an emma mets";
+    };
+
+
+  };
+
+  describe "HTFeed::PackageType::EMMA::METS" => sub {
+    my $stage;
+
+    before each => sub {
+
+      mock_zephir();
+
+      HTFeed::PackageType::EMMA::Unpack->new(volume => $volume)->run();
+      HTFeed::PackageType::EMMA::SourceMETS->new(volume => $volume)->run();
+      HTFeed::Stage::Pack->new(volume => $volume)->run();
+
+      # expire cached info in volume
+      $volume = HTFeed::Volume->new(namespace => 'emma',
+        objid => $objid,
+        packagetype => 'emma');
```

Comment:
Looks like changing the namespace to 'test' clears up many of the errors in these tests. Otherwise they really don't get off the ground due to "Invalid packagetype emma for namespace test" complaint. I think I need to get the EMMA testsuite a little further along in order to see what's happening in the ClamAV integration.

Reply:
OK -- I think I must have files that I haven't pushed or there's some submodule nonsense going on here. Let me see what I can do.

Reply:
That latest commit worked, and was similar enough to my workarounds that it was an easy merge.